src/components/ContactsView.js

```diff
@@ -1,23 +1,36 @@
-import { useEffect, useState } from "react"
-import { useParams } from "react-router-dom"
+import { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
 
 function ContactsView() {
-  const [contact, setContact] = useState(false)
+  const [contact, setContact] = useState(false);
 
   //TODO: Get the contact to load from the params and fetch.
   //With useEffect, load the contact when params changes
   //and update contact state
+  const params = useParams();
+
+  const { id } = params;
+
+  useEffect(async () => {
+    const data = await fetch(`http://localhost:4000/contacts/${id}`);
+    const contact = await data.json();
+    setContact(contact);
+  }, [id]);
 
   if (!contact) {
-    return <p>Loading</p>
+    return <p>Loading</p>;
   }
 
   return (
     <div>
-      <h2>{contact.firstName} {contact.lastName}</h2>
-      <p>{contact.street} {contact.city}</p>
+      <h2>
+        {contact.firstName} {contact.lastName}
+      </h2>
+      <p>
+        {contact.street} {contact.city}
+      </p>
     </div>
-  )
+  );
 }
 
-export default ContactsView
\ No newline at end of file
+export default ContactsView;
```

Comment:
well organised core criteria, would be able to easier add functionality/extensions if you want to spend some extra time on it